siliconcompiler/flows/fpgaflow.py

```diff
@@ -103,108 +98,6 @@ def setup(chip, flowname='fpgaflow'):
     return flow
 
 
-##################################################
-def flow_lookup(partname):
```

Comment:
Why was the flow lookup deleted?

Reply:
It has essentially been replaced by the if/else logic setting the flowpipe variable.  This allows things to be a little more a la carte between flows and targets.  There's an open philosophical debate to have about separation of concerns for FPGA flows between flows and targets that the current code largely defers by only supporting open source flows for VPR and nextpnr and only for specific demo architectures (ice40 and my academic example for VPR).  The priority has been given to supporting a subset of what was originally envisioned that can be tested in CI, rather than including code that cannot be tested due to vendor-specific SW requirements.  Followup dicussions recommended to scope work beyond the reach of this PR.

Reply:
This seems to break the existing functionality? There is a reason why I made the flow somwehat complex. I wanted to be able to use one entry point, with the part name dicating what path to take. Of course we can always specify separate flows and targets, but this is a bigger mental load.

Reply:
In some cases, breaking existing functionality was intentional.  To points made by @gadfort when the two of us were comparing notes, there are limits to what SC can support w/respect to commercial FPGA RTL to bitstream flows because dependencies are introduced on tools (e.g. Vivado) that come with terms of use that limit their open adoption.  Additionally, the Venn diagram of tools supported by yosys + (VPR+genfasm || nextpnr) is basically ICE40 and (maybe?) Xilinx 7 series.  I wanted to solve the minimum viable problem that put a new flow back on SC main with minimal development effort, even if it meant a temporary step backwards that pushed us to create new PRs to re-introduce things that worked at point (notably, some limited support for Xilinx).  

As for the "one entry point set by part number" vs. a more "build your flow/target pair for a given FPGA family / CAD flow pair" approach, I'm not 100% married to anything.  Part of how I arrived at the current approach was simply trying to develop a mental model for what it meant to be a flow and what it meant to be a target.  In the ASIC arena, SC only has one tool chain to worry about (OpenROAD and friends) and many process technologies + associated cell libraries.  For the FPGA case, there are two open source tool chains, and (theoretically) many families of parts to program.  Following the analogy that flow == ASIC tool chain == FPGA tool chain and target == ASIC process == FPGA parts family, I arrived at what you see here.

Reply:
Got it, thanks for describing the process. My view is that we should have as few entry points as possible to reduce the mental load on the user to memorize arbitrary names. In the case of FPGAs, whether using vpr, vivado, quartus we always have to specify which partname we are targeting.

I feel strongly that we should simplify things for the user as much as possible and it is my view that this is done by limiting the introuction of new things. This is why the fpgaflow. Can you restore the previous logic while enabling the new target?